src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterStep.cs

```diff
@@ -74,6 +75,7 @@ public class PackageUpdaterStep : UpgradeStep
             _packageRestorer = packageRestorer ?? throw new ArgumentNullException(nameof(packageRestorer));
             _packageAnalyzers = packageAnalyzers ?? throw new ArgumentNullException(nameof(packageAnalyzers));
             _analysisState = null;
+            _packageAnalyzer = new PackageAnalyzer(_packageRestorer, _packageAnalyzers, logger);
```

Comment:
This should be injected in rather than new'd up here. You can register it in the PackageUpdaterStepExtensions